src/outagesim/backup_reliability.jl

```diff
@@ -240,60 +466,69 @@ If ``marginal_survival`` = true then result is chance of surviving in given outa
+julia> survival_over_time_gen_only(critical_load=critical_load_kw, gen_operational_availability=gen_operational_availability, 
+                                failure_to_start=failure_to_start, failure_to_run=failure_to_run, num_generators=num_generators, 
+                                gen_capacity_kw=gen_capacity_kw, max_duration=max_duration, marginal_survival = true)
 4×3 Matrix{Float64}:
  0.96  0.4096  0.761856
  0.64  0.8704  0.761856
  0.96  0.8704  0.761856
  0.96  0.8704  0.262144
 
-julia> survival_over_time_gen_only(critical_load, OA, failure_to_start, failure_to_run, num_generators, gen_capacity, max_duration; marginal_survival = false)
+julia> survival_over_time_gen_only(critical_load=critical_load_kw, gen_operational_availability=gen_operational_availability, 
+                                failure_to_start=failure_to_start, failure_to_run=failure_to_run, num_generators=num_generators, 
+                                gen_capacity_kw=gen_capacity_kw, max_duration=max_duration, marginal_survival = false)
 4×3 Matrix{Float64}:
  0.96  0.4096  0.393216
  0.64  0.6144  0.557056
  0.96  0.8704  0.761856
  0.96  0.8704  0.262144
 ```
 """
-                     
-function survival_over_time_gen_only(critical_load::Vector, OA::Real, failure_to_start::Real, failure_to_run::Real, num_generators::Int, 
-    gen_capacity::Real, max_duration::Int; marginal_survival = true)::Matrix{Float64} 
-
-    t_max = length(critical_load)
+function survival_over_time_gen_only(;
+    critical_load_kw::Vector, 
+    gen_operational_availability::Union{Real, Vector{<:Real}}, 
+    failure_to_start::Union{Real, Vector{<:Real}}, 
+    failure_to_run::Union{Real, Vector{<:Real}},
+    num_generators::Union{Int, Vector{Int}}, 
+    gen_capacity_kw::Union{Real, Vector{<:Real}},
+    max_duration::Int,
+    marginal_survival = true)::Matrix{Float64} 
+
+    t_max = length(critical_load_kw)
     #
-    generator_production = collect(0:num_generators).*gen_capacity
+    generator_production = generator_output(num_generators=num_generators, gen_capacity_kw=gen_capacity_kw) 
     #Initialize lost load matrix
     survival_probability_matrix = zeros(t_max, max_duration)
     #initialize amount of extra generation for each critical load hour and each amount of generators
-    generator_markov_matrix = markov_matrix(num_generators, failure_to_run)
+    generator_markov_matrix = markov_matrix(num_gen=num_generators, fail_prob=failure_to_run)
   
     #Get starting generator vector
-    starting_gens = starting_probabilities(num_generators, OA, failure_to_start) #initialize gen battery prob matrix
+    starting_gens = starting_probabilities(num_gen=num_generators, gen_operational_availability=gen_operational_availability, gen_failure_to_start=failure_to_start) #initialize gen battery prob matrix
 
-    #start loop
     for t  = 1:t_max
         gen_probs = starting_gens
         #
```

Comment:
were you intending to add something here?

Reply:
and line 532

Reply:
Removed the dangling #'s